exporters/variables-scss/src/content/generators/cssGenerator.ts

```diff
@@ -0,0 +1,93 @@
+import { DimensionToken, StringToken, Token, TokenGroup, TokenType } from '@supernovaio/sdk-exporters';
+import { CSSHelper } from '@supernovaio/export-helpers';
+import { toPlural } from '../helpers/stringHelper';
+import { tokenVariableName } from '../helpers/tokenHelper';
+
+export const tokenToCSSByType = (
+  token: Token,
+  mappedTokens: Map<string, Token>,
+  tokenGroups: Array<TokenGroup>,
+  withParent: boolean,
+): string | null => {
+  if (token.tokenType === TokenType.dimension) {
+    const dimensionToken = token as DimensionToken;
+    const name = tokenVariableName(dimensionToken, tokenGroups, withParent);
+    const value = dimensionToken.value?.measure;
+    const unit = CSSHelper.unitToCSS(dimensionToken.value?.unit);
+
+    return `$${name}: ${value}${unit} !default;`;
+  }
+
+  if (token.tokenType === TokenType.string) {
+    const stringToken = token as StringToken;
+    const name = tokenVariableName(stringToken, tokenGroups, withParent);
+    const value = stringToken.value.text;
+
+    return `$${name}: ${value} !default;`;
+  }
+
+  return null;
+};
+
+export const generateCssFromTokens = (
+  tokens: Token[],
+  mappedTokens: Map<string, Token>,
+  tokenGroups: Array<TokenGroup>,
+  hasParentPrefix: boolean,
+): string => {
+  return tokens
+    .map((token) => tokenToCSSByType(token, mappedTokens, tokenGroups, hasParentPrefix))
+    .filter(Boolean)
+    .join('\n');
+};
+
+const generateObjectContent = (tokens: Array<Token>, tokenGroups: Array<TokenGroup>, withParent: boolean): string => {
+  let result = '';
+
+  tokens.forEach((token) => {
+    const name = tokenVariableName(token, tokenGroups, withParent);
+    const numericPart = name.match(/\d+/)?.[0];
+    const prefix = `${token.origin?.name?.split('/')[0].toLowerCase()}-`;
+    const nonNumericPart = name.replace(prefix, '');
+    if (numericPart) {
+      result += `${numericPart}: $${name},\n`;
+    } else if (nonNumericPart) {
+      result += `${nonNumericPart}: $${name},\n`;
+    }
+  });
+
+  return result;
+};
+
+export const generateCssObjectFromTokens = (
+  tokens: Array<Token>,
+  mappedTokens: Map<string, Token>,
+  tokenGroups: Array<TokenGroup>,
+  hasParentPrefix: boolean,
+): string | null => {
+  // Create a map where the key is the origin name and the value is an array of tokens
```

Comment:
Based on this comment it looks like a `map` function should be used. Because you are mapping an old structure to a new one.

Reply:
But in this case you don't want to create a new array, but update the originNameMap based on the values ​​in tokens. Therefore, using forEach is IMHO more appropriate here.

Reply:
But you are creating a new object with values. You are starting with an empty Map (that's the accumulator of the values).

Maybe the `map` is not a good idea here and `reduce` will be better.

As I understand it correctly, you are creating something like:

```js
{
  'token-name': [Token]
}
```
?

The test for this function would be helpful.

Reply:
In general, tests would need to be written. As part of that issue or will we create another one?

Reply:
Maybe a part of it will be great, but not all.